### PR TITLE
Add proof-size related benchmarks to message lane module

### PR DESCRIPTION
**Pull request by svyatonik**
part of #670 - while I still have no solution for described issue, having these benchmarks may give information on how proof size affects tx weight.

This PR also fixes issues with existing benchmarks that were broken after weight formulas were introduces.

### CODE REVIEW

File: modules/message-lane/src/benchmarking.rs
Comment (HCastano) [inline modules/message-lane/src/benchmarking.rs:43]:
```suggestion
	/// included in the proof.
```

File: modules/message-lane/src/benchmarking.rs
Comment (HCastano) [inline modules/message-lane/src/benchmarking.rs:39]: I don't quite understand the difference between `HasExtraNodes` and `HasLargeLeaf` - can you give me an example or something to help clarify?

File: modules/message-lane/src/benchmarking.rs
Comment (HCastano) [inline modules/message-lane/src/benchmarking.rs:579]: Just wondering, why don't you need a benchmark with`ProofSize::Minimal`?

File: modules/message-lane/src/lib.rs
Comment (HCastano) [inline modules/message-lane/src/lib.rs:113]: Why are we changing `From<u32>` to `From<u64>`?

**Review by HCastano — APPROVED** [review]
Looks alright, but since I'm unfamiliar with the storage proof stuff I'd get another pair of eyes on this before merging

File: modules/message-lane/src/benchmarking.rs
Comment (svyatonik) [reply modules/message-lane/src/benchmarking.rs:39]: Imagine binary tree and operation `binary_tree.encode()` that avoids any optimizations and encodes nodes, not values - i.e. pointers are also included in in the encoding. So there are two kinds of grow: (1) simply add 1 large value to the tree => since we include values into encoding, encoded value will be larger (2) insert many 0-byte values => number of nodes will grow and since we put internal information in the encoding, encoded value will be larger. So `HasExtraNodes` is the latter and `HasLargeLeaf` is the former. With these benchmarks, you may see that extra-nodes way has larger impact on performance, but the difference is not that big.

File: modules/message-lane/src/benchmarking.rs
Comment (svyatonik) [reply modules/message-lane/src/benchmarking.rs:579]: All other benchmarks are running with `Minimal` :)

File: modules/message-lane/src/lib.rs
Comment (svyatonik) [reply modules/message-lane/src/lib.rs:113]: The payment for message in some benchmarks doesn't fit `u32` :/ As I noted in the description, right now benchmarks are broken and this is a fix

**Review by tomusdrw — APPROVED** [review]
lgtm!

**Pull request merged**
